## Задание 1
Первое, что мы можем сделать - это посчитать описательные статистики по лайкам и просмотрам.

Укажите, что из этого НЕ ХАРАКТЕРНО для наших данных. Возможно, вы догадаетесь о правильном ответе даже без описательных статистик :)

- [ ] По мерам центральной тенденции просмотры опережают лайки
- [ ] Медиана лайков приблизительно такая же, как и среднее по ним же
- [ ] Самые популярные посты собирают миллионы просмотров

In [3]:
import pandas as pd
from read_db import Getch

### По мерам центральной тенденции просмотры опережают лайки
Будем считать по дням.  
МЦТ: мода, медиана, среднее.   
Думаю, что тут ничего не надо считать, так как понятно, что просмотров больше лайков, и, следовательно, среднее, мода и медиана у просмотров будет больше, чем у лайков.

Ответ: этот вариант не нужно выбирать

In [11]:
view_query = """
SELECT
    toDate(time) date,
    count(distinct user_id) count_users
FROM simulator_20220420.feed_actions
WHERE action = 'view'
GROUP BY toDate(time)
"""

In [12]:
view_data = Getch(view_query).df

In [13]:
view_data.head(5)

date  count_users
0 2022-03-08         1198
1 2022-03-09         2325
2 2022-03-10         3388
3 2022-03-11         4121
4 2022-03-12         4917

In [14]:
view_data.describe()

count_users
count     57.000000
mean   12548.666667
std     4699.846070
min     1198.000000
25%     9637.000000
50%    13173.000000
75%    16079.000000
max    19129.000000

In [18]:
view_data['count_users'].value_counts()

1198     1
13201    1
14539    1
14774    1
14638    1
14218    1
14688    1
15071    1
15758    1
15963    1
16422    1
19123    1
15609    1
16079    1
16781    1
16968    1
17671    1
17960    1
17369    1
17132    1
14073    1
18248    1
18666    1
18608    1
19129    1
18613    1
18572    1
13928    1
12968    1
2325     1
12651    1
3388     1
4121     1
4917     1
5245     1
5281     1
5996     1
6970     1
7743     1
8373     1
8870     1
8855     1
8865     1
9637     1
9926     1
10600    1
10717    1
11388    1
11188    1
11035    1
11373    1
11901    1
12478    1
12780    1
13173    1
12829    1
14682    1
Name: count_users, dtype: int64

### Медиана лайков приблизительно такая же, как и среднее по ним же

In [4]:
likes_query = """
SELECT
    toDate(time) date,
    count(distinct user_id) count_users
FROM simulator_20220420.feed_actions
WHERE action = 'like'
GROUP BY toDate(time)
"""

In [5]:
likes_data = Getch(likes_query).df

In [6]:
likes_data.head(5)

date  count_users
0 2022-03-08          844
1 2022-03-09         2251
2 2022-03-10         3344
3 2022-03-11         4063
4 2022-03-12         4851

In [7]:
likes_data.describe()

count_users
count     57.000000
mean   12301.210526
std     4610.210457
min      844.000000
25%     9526.000000
50%    12740.000000
75%    15909.000000
max    18769.000000

Среднее = 12300.21, медиана = 12740.   
Ответ: этот вариант не нужно выбирать.
UPD. Оказалось, что это вариант нужно выбрать, потому что они не равны.

In [20]:
likes_data['count_users'].value_counts()

844      1
12740    1
14260    1
14573    1
14451    1
14070    1
14559    1
14872    1
15421    1
15595    1
16020    1
18769    1
15313    1
15909    1
16483    1
16633    1
17385    1
17700    1
16957    1
16899    1
13831    1
17902    1
18319    1
18206    1
18592    1
18170    1
17879    1
13585    1
12580    1
2251     1
12418    1
3344     1
4063     1
4851     1
5131     1
5163     1
5880     1
6781     1
7628     1
8264     1
8765     1
8751     1
8782     1
9526     1
9753     1
10331    1
10497    1
11154    1
10950    1
10846    1
11137    1
11759    1
12277    1
12655    1
13045    1
12642    1
14008    1
Name: count_users, dtype: int64

### Самые популярные посты собирают миллионы просмотров

In [8]:
# Посмотрим топ-10 постов по просмотрам за все время
top_10_posts_query = """
SELECT
    post_id,
    count(user_id) count_views
FROM simulator_20220420.feed_actions
WHERE action = 'view'
GROUP BY post_id
ORDER BY count_views desc
LIMIT 10
"""

In [9]:
top_10_posts_data = Getch(top_10_posts_query).df

In [10]:
top_10_posts_data

post_id  count_views
0     3192        89909
1     3246        76393
2     1930        75750
3     3795        75448
4     3689        70686
5     2287        67761
6     2235        66728
7     3172        64376
8     3238        62093
9     3538        60968

На миллион не похоже :)    

Ответ: выбираем этот вариант